src/arrangement/core.cljc

```diff
@@ -78,10 +90,11 @@
         (compare-seqs rank a b)
 
         :else
-        (let [class-diff (compare (.getName (class a))
-                                  (.getName (class b)))]
+        (let [class-diff (compare (type-name a)
+                                  (type-name b))]
           (if (zero? class-diff)
-            (if (instance? java.lang.Comparable a)
-              (compare a b)
-              (compare (str a) (str b)))
```

Comment:
btw in an original version of this you had `pr-str` rather than `str` - was there a reason you changed these?

I'm thinking of something like this:
```clojure
user=> (str (lazy-seq))
"clojure.lang.LazySeq@1"
user=> (pr-str (lazy-seq))
"()"
```